subsys/usb/device_next/usbd_device.c

```diff
@@ -263,9 +263,17 @@ int usbd_enable(struct usbd_context *const uds_ctx)
 		goto enable_exit;
 	}
 
+	uds_ctx->status.enable_suspended = false;
 	ret = udc_enable(uds_ctx->dev);
 	if (ret != 0) {
-		LOG_ERR("Failed to enable controller");
+		if (ret == -EWOULDBLOCK) {
+			/* The enable operation will be retried on VBUS detected event. */
```

Comment:
This mixes the responsibility for what code is supposed to enable the USB stack.

Reply:
I want to make the `usbd_enable` operation available even when the USB cable is disconnected. Do you see any better way to do that?

Reply:
Maybe I could somehow forward the UDC_EVT_VBUS_READY event to usbd_device.c module so that it retries the enable operation,

Reply:
But why would the stack have to retry the enable operation if it returned `-EWOULDBLOCK`. By the time `-EWOULDBLOCK` is returned the handling should be over.

Reply:
From the application perspective I'd like to make it possible to call `usbd_enable` even if the cable is disconnected. The idea is that the USB stack will remember that the `usbd_enable` has been called and retry it when the cable is connected.

Reply:
Oh, I see it's problematic with how the `usbd_disable`, `usbd_shutdown` work.

Reply:
If you don't see any good solution, I can simplify it and leave just Kconfig option that will allow for not blocking the `usbd_enable` call forever and return error instead. Application would have to retry the enable operation on its own.

Reply:
It'd look similar to this: https://github.com/alstrzebonski/zephyr/commit/043d47c6472059109dcec8d32b264cf2b59f629c

Reply:
Application retrying enable on its own is the only sensible solution. Whether or not `-EWOULDBLOCK` is appropriate is a separate matter.

Reply:
Okay, I will limit this PR to just returning `-EWOULDBLOCK` on timeout.